examples/chat/README.md

```diff
@@ -51,9 +51,9 @@ client pointer from the `clients` map, the hub closes the clients's `send`
 channel to signal the client that no more messages will be sent to the client.
 
 The hub handles messages by looping over the registered clients and sending the
-message to the client's `send` channel. If the client's `send` buffer is full,
-then the hub assumes that the client is dead or stuck. In this case, the hub
-unregisters the client and closes the websocket.
+json marshalled message to the client's `send` channel. If the client's `send` 
```

Comment:
Can you expand the docs a bit more? I'm happy to merge this, but this is a good opportunity to improve the docs for newbies. 

* explain the Message type more
* reinforce the single-writer / no concurrent writer requirement at all points (very common mistake)
* make it clear that JSON is an option, and used for the example as it represents a common serialization for WebSocket messages

Reply:
Sure, I can do this. I'm not sure what to say about the no concurrent writer requirement though.

Reply:
I'll try to have it done this week

Reply:
@elithrar I update the documentation in [the latest commit](https://github.com/gorilla/websocket/pull/621/commits/d18f90cf8f1b6ef5cb40d666aacff295d8a1c594).